homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -99,6 +100,7 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
```

Comment:
Per [ADR-0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md), we no longer allow integrations to be configured via YAML – they must be configured via the UI. We allow existing YAML configuration to persist, but not altering that configuration.

So, the way forward is to add UI configuration. This platform seems like it would be a great helper, so that would be a route to consider.

Reply:
That is my long term plan as per https://community.home-assistant.io/t/the-future-of-the-bayesian-integration-thoughts-wanted/465476/7 and unqiue_id generation and handling will be part of that.

For now this would not be a breaking change, and we have already changed config in #67631

Reply:
"To protect project goals and to provide clarity to our users and contributors, we’re introducing the following rules on how integrations need to be configured:

- Integrations that communicate with devices and/or services are only configured via the UI. In rare cases, we can make an exception.
- All other integrations are configured via YAML or via the UI.

These rules apply to all new integrations. **Existing integrations that should not have a YAML configuration**, are allowed and encouraged to implement a configuration flow and remove YAML support. Changes to existing YAML configuration for these same existing integrations, will no longer be accepted."

According to ADR-0010 this only applies to integrations "that should not have a YAML configuration ". I don't think Bayesian is one of those?

Reply:
Right this falls under the “Integrations that process Home Assistant data and make this available to other integrations. Examples are template, stats, derivative, utility meter.” case where yaml is still allowed, and adding UI configuration is optional (but encouraged). This is fundamentally a helper integration, which have retained yaml support when added to the UI. (ADR0010 as written does not technically prohibit removing yaml support, but the way balloob and others have talked about this, it makes it sound like it is not expected to occur.

It is ADR007 that could be a concern here, since the yaml config is still under the platform keys, rather than being under the integration domain key, meaning we technically should not be changing the config schema.

Unfortunately fixing that would be another breaking change for this integration (albeit one that can have a deprecation window where the old config continues working for a few releases). It is undesireable to have multiple breaking changes for an integration that would be expected to impact most users of said integration in such short succession.